### DISCUSSION TITLE
Line Numbers missing in Exceptions

### DISCUSSION THREAD

**Discussion by goswinr** [dotnet/fsharp#13293] [Q&A] [answered]
When using F# Interactive via FSharp.Compiler.Service 41.0.4 on `net48` Exceptions do not include line numbers.
Line numbers are only shown on `net6.0` : 
![image](https://user-images.githubusercontent.com/884885/173533784-730c2136-c48f-4afb-887f-9ac83f92d285.png)

Is there a way to make exceptions include line numbers when on net48 ?
I am hosting fsi in a much larger net48 process. So I can't switch to net6.0.

code used: 
```fsharp
let code = """
try
    8 / 0 |> ignore
with e -> 
    printfn $"Runtime: {System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription}"
    printfn $"Error: {e}" 
"""

open System
open System.IO
open FSharp.Compiler.Interactive.Shell

let eval(codeStr) = 
    let inn = new StringReader("")
    let out = Console.Out
    let config = FsiEvaluationSession.GetDefaultConfiguration()
    let noArgs = [| "the first arg is neded, but ignored"|]
    let session = FsiEvaluationSession.Create(config, noArgs, inn, out, out)
    session.EvalInteraction(codeStr)

eval(code)
    
System.Console.ReadLine() |> ignore
```
and 

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net48;net6.0</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="Program.fs" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Update="FSharp.Core" Version="6.0.4" />
    <PackageReference Include="FSharp.Compiler.Service" Version="41.0.4" />
  </ItemGroup>
</Project>
```

full repo: https://github.com/goswinr/ErrorLineNumberMissing

**Comment by dsyme** [comment]
You might try passing `--debug` in the arguments to the interactive session

**Comment by dsyme** [comment]
You might also try `--multiemit`

**Reply by goswinr** [reply]
thanks @dsyme` --multiemit` fixed it ! 
![image](https://user-images.githubusercontent.com/884885/173605318-754f2052-a781-460b-afc7-a37be57c9c93.png)
The documentation here is also great: https://fsharp.github.io/fsharp-compiler-docs/fsi-emit.html

**Comment by dsyme** [comment]
I'll convert this to a discussion since I don't think we'd treat this as a specific bug yet based on the info above

**Comment by goswinr** [comment]
@dsyme I still have the problem when hosting a fsi session in a net48 process. 
I tried changing the current directory [like in this other hosting problem ](https://github.com/dotnet/fsharp/issues/11770) but it didn't help. 
The path to `input.fsx` and the line number don't show.
(In standalone line numbers show  with `--multiemit`)

Do you have any suggestions? Or can you suggest places to start looking for the issues in the source?

**Reply by dsyme** [reply]
Hmmm I'm not really sure what to look at next.   Could you link your latest attempt?

**Comment by goswinr** [comment, marked as answer]
I am running FCS in net8 now, inside Rhino, that fixed it for me.
